cluster-autoscaler/utils/test/test_utils.go

```diff
@@ -67,6 +67,15 @@ func BuildTestPod(name string, cpu int64, mem int64) *apiv1.Pod {
 	return pod
 }
 
+// BuildDSTestPod creates a DaemonSet pod with cpu and memory.
+func BuildDSTestPod(name string, cpu int64, mem int64) *apiv1.Pod {
```

Comment:
Helpers function in this file and actuator_test are similar. I am not sure why we didn't modify the functions here to use in actuator_test.go (maybe we didn't want to introduce dependency?). I have kept them separate for now i.e., I have added a new helper function for DS in actuator_test.go and introduced a new helper function for DS here.

Reply:
I am not sure if merging the helper functions should be done in this PR but any feedback is welcome :pray:

Reply:
I think it would be nice to leverage [option pattern](https://golang.cafe/blog/golang-functional-options-pattern.html) in this file, so that tests can only set pod fields they care about:

```
BuildTestPod("pod", withCpu(2), withMemoryMb(123), withDaemonSetOwnerRef("some-uid"))
```

That would make this more readable and maintainable vs current approach, which duplicates existing code. This file should only really depend on k8s and be importable from other places like actuator tests.

Reply:
@x13n option pattern is very interesting. Thank you for telling me about it. If you have any suggestions like these again, please tell me all about it :D  

As for implementing it, I think it might be better to do it in a separate PR. This is because `BuildTestPod` [is used in a lot of places](https://github.com/search?q=repo%3Akubernetes%2Fautoscaler%20BuildTestPod&type=code). 
Current PR is big as is. Making changes in the test utils would increase the PR size further (which might make it harder to review). Also, I think we could replace many functions in [test_utils.go](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/utils/test/test_utils.go) for both pods and nodes. I have started a separate PR to refactor test utils using options pattern: https://github.com/kubernetes/autoscaler/pull/5932

Another option is to create a separate function which wraps BuildTestPod and uses options pattern but I think that would mean we end up creating a function very similar to `BuildTestPod` (which can be a potential source of confusion). 

Let me know what you think.

Reply:
Separate PR makes a lot of sense, thanks!